### PR TITLE
extend TimerOutput to have dynamic Section width

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/dealii/dealii/issues/6691

### CODE REVIEW

File: tests/base/timer_08.cc [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+void
+test(TimerOutput::OutputType output_type)
+{
+  std::stringstream ss;
+  {
+    TimerOutput t(ss, TimerOutput::summary, output_type);
+
+    t.enter_subsection("hi");
+    burn(50);
```
Comment (masterleinad) [inline tests/base/timer_08.cc]: Do we actually need that?

File: tests/base/timer_08.cc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    t.leave_subsection("hi");
+
+    t.enter_subsection(
+      "this is a very long section name that previously did not work");
+    burn(50);
```
Comment (masterleinad) [inline tests/base/timer_08.cc]: see above

File: tests/base/timer_08.cc (hunk 1 above)
Comment (davydden) [reply tests/base/timer_08.cc]: probably not, will remove.

Comment (davydden) [reply tests/base/timer_08.cc]: I checked, we do need this.

Comment (masterleinad) [reply tests/base/timer_08.cc]: I see. We need to make sure that we are not using scientific notation for this to work.
May I suggest to use
```
     std::this_thread::sleep_for(std::chrono::seconds(1));
```
instead?

**Comment by masterleinad** [comment]
I also recognized just now, that you are using
```
std::ofstream logfile("output");
deallog.attach(logfile);
```
instead of
```
initlog();
```
Would you mind to fix this pet peeve of mine as well?

File: tests/base/timer_08.cc (hunk 1 above)
Comment (davydden) [reply tests/base/timer_08.cc]: > I see. We need to make sure that we are not using scientific notation for this to work.

correct. The filtering of timing from the table is primitive and does not work with scientific notation.

Will try `sleep_for()`...

Comment (davydden) [reply tests/base/timer_08.cc]: @masterleinad sleep_for does not work.

Comment (masterleinad) [reply tests/base/timer_08.cc]: For me, that works. Still, let's not bother too much. Thanks for trying!

**Comment by davydden** [comment]
@masterleinad are we good to merge?

**Comment by masterleinad** [comment]
Yes, I approved this from the beginning. I guess this is quite uncontroversial.

**Pull request merged**